File: content/problems/3038-maximum-number-of-operations-with-the-same-score-i/solutions/scan-fixed-adjacent-pairs/solution.cpp
class Solution {
    public:int maxOperations(vector<int>& nums){
        int target=nums[0]+nums[1],answer=0;
        for(int i=0;i+1<(int)nums.size();i+=2){
            if(nums[i]+nums[i+1]!=target)break;
            ++answer;
        }
        return answer;
    }
};

File: content/problems/3038-maximum-number-of-operations-with-the-same-score-i/solutions/scan-fixed-adjacent-pairs/Solution.kt
class Solution {
    fun maxOperations(nums:IntArray):Int{
        val target=nums[0]+nums[1];
        var answer=0;
        var index=0;
        while(index+1<nums.size){
            if(nums[index]+nums[index+1]!=target)break;
            answer++;
            index+=2
        };
        return answer
    }
}
